app/controllers/courses/watchlist_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/watchlist_edits"
+
+class Courses::WatchlistController < ApplicationController
+  include CourseHelper
+
+  # /courses/:slug/students/add_to_watchlist
+  def add_to_watchlist
+    course = find_course_by_slug(params[:slug])
+    student_names = CoursesUsers.new.user_page(course.students)
+    WatchlistEdits.new(course.home_wiki, student_names).oauth_credentials_valid?(current_user)
```

Comment:
This is not a great interface, as it does not communicate what it's going to do.

Here are some interfaces that would make sense to me:

`WatchlistEdits.new(wiki, current_user).add_to_watchlist(array_of_pagenames)`

`WatchlistEdits.new(wiki, current_user).watch_userpages(array_of_users)`

`WatchUserpages.new(wiki:, current_user:, users:)`

Reply:
Okay